app/main/views/audits.py

```diff
@@ -180,6 +212,52 @@ def acknowledge_audit(audit_id):
     return jsonify(auditEvents=audit_event.serialize()), 200
 
 
+@main.route('/audit-events/<int:audit_id>/acknowledge-including-previous', methods=['POST'])
+def acknowledge_including_previous(audit_id):
+    updater_json = validate_and_return_updater_request()
+
+    audit_event = AuditEvent.query.get(audit_id)
+    if audit_event is None:
+        abort(404, "No audit event with this id")
+
+    result = db.session.execute(AuditEvent.__table__.update().returning(
+        AuditEvent.__table__.c.id
+    ).where(db.and_(
+        AuditEvent.__table__.c.object_id == audit_event.object_id,
+        AuditEvent.__table__.c.object_type == audit_event.object_type,
+        AuditEvent.__table__.c.type == audit_event.type,
+        AuditEvent.__table__.c.acknowledged == db.false(),
+        # ugly, but this just implements the same "id tie breaker" behaviour for created_at-equal events as the
+        # ordering we use in list_audits. this way there is some consistency between the two views as to what events
+        # are considered "previous" in such cases. we could use postgres composite types to express this far more
+        # neatly, but i can't get sqlalchemy to work with anonymous composite types.
```

Comment:
I don't understand, can't the order just be "date then id". Once you've hit the primary / any candidate key, no further ordering is necessary?